File: crates/psychevo-runtime/src/agents/tests/policy_and_control/fixtures/child_hook_plugin_worker.py
#!/usr/bin/env python3
import json
import os
import pathlib
import sys


for line in sys.stdin:
    request = json.loads(line)
    method = request.get("method")
    if method == "initialize":
        result = {"ok": True}
    elif method == "contributions/list":
        result = {"tools": []}
    elif method == "hooks/call":
        data = pathlib.Path(os.environ["PSYCHEVO_PLUGIN_DATA"])
        data.mkdir(parents=True, exist_ok=True)
        with (data / "child-hook.jsonl").open("a", encoding="utf-8") as handle:
            handle.write(
                json.dumps(
                    {
                        "event": request.get("params", {})
                        .get("hook", {})
                        .get("event")
                    }
                )
                + "\n"
            )
        result = {"feedback": "plugin child hook ran"}
    elif method == "shutdown":
        result = {"ok": True}
    else:
        result = {}
    print(
        json.dumps({"jsonrpc": "2.0", "id": request.get("id"), "result": result}),
        flush=True,
    )

File: crates/psychevo/src/agents/tests/policy_and_control/fixtures/child_hook_plugin_worker.js
#!/usr/bin/env node
const fs = require("fs");
const path = require("path");
const readline = require("readline");

readline.createInterface({ input: process.stdin, crlfDelay: Infinity }).on("line", (line) => {
  const request = JSON.parse(line);
  let result = {};
  if (request.method === "initialize" || request.method === "shutdown") {
    result = { ok: true };
  } else if (request.method === "contributions/list") {
    result = { tools: [] };
  } else if (request.method === "hooks/call") {
    const data = process.env.PSYCHEVO_PLUGIN_DATA;
    fs.mkdirSync(data, { recursive: true });
    fs.appendFileSync(
      path.join(data, "child-hook.jsonl"),
      `${JSON.stringify({ event: request.params?.hook?.event })}\n`,
      "utf8",
    );
    result = { feedback: "plugin child hook ran" };
  }
  process.stdout.write(`${JSON.stringify({ jsonrpc: "2.0", id: request.id, result })}\n`);
});
